code/modules/projectiles/guns/projectile/launchers.dm

```diff
@@ -83,3 +83,749 @@
+				
+			if(user.get_inactive_hand())
+				to_chat(user, "<span class='warning'>Your other hand needs to be free to deploy [src].</span>")
+			else
+				to_chat(user, "<span class='notice'>You begin to deploy [src].</span>")
+				if(do_after(user, handle_time, TRUE, target = user))
+					weapon_stage = DEPLOYED
+					update_weapon()
+					user.client.mouse_pointer_icon = initial(user.client.mouse_pointer_icon)
+					playsound(user, 'sound/weapons/gun_interactions/selector.ogg', 75, 0)
+					user.visible_message("<span class='notice'>[user] deploys [src].</span>", \
+											"<span class='notice'>You deploy [src].</span>")
+				else
+					to_chat(user, "<span class='warning'>You stop deploying [src].</span>")
+			deploy_lock = FALSE
+			
+		if(DEPLOYED) // Setup up to ready to fire.
+			deploy_lock = TRUE
+			if(user.get_inactive_hand())
+				to_chat(user, "<span class='warning'>Your other hand needs to be free to ready [src].</span>")
+			else
+				to_chat(user, "<span class='notice'>You brace to fire [src].</span>")
+				if(do_after(user, handle_time, TRUE, target = user))
+					weapon_stage = FIRE_READY
+					update_weapon()
+					user.client.mouse_pointer_icon = file("icons/mecha/mecha_mouse.dmi")
+					playsound(user, 'sound/weapons/targeton.ogg', 75, 0)
+					user.visible_message("<span class='notice'>[user] is ready to fire [src].</span>", \
+											"<span class='notice'>[src] is ready to fire.</span>")
+				else
+					to_chat(user, "<span class='warning'>Your grip is broken deploying the [src].</span>")
+			deploy_lock = FALSE
+				
+		if(FIRE_READY) // Setup back to deployed.
+			weapon_stage = DEPLOYED
+			update_weapon()
+			user.client.mouse_pointer_icon = initial(user.client.mouse_pointer_icon)
+			playsound(user, 'sound/weapons/targetoff.ogg', 75, 0)
+			user.visible_message("<span class='notice'>[user] lowers [src].</span>", \
+									"<span class='notice'>You lower [src].</span>")
+
+		if(SPENT)
+			to_chat(user, "<span class='notice'>[src] is deplited.</span>")
+			return
+			
+/obj/item/gun/projectile/S72LRW/dropped(mob/user)
+	if(weapon_stage == FIRE_READY)
+		weapon_stage = DEPLOYED
+		update_weapon()
+		user.client.mouse_pointer_icon = initial(user.client.mouse_pointer_icon)
+		playsound(user, 'sound/weapons/targetoff.ogg', 75, 0)
+		to_chat(user, "<span class='warning'>You drop [src].</span>")
+
+/obj/item/SLRWpin/Initialize()
+	name = "SLRW safety pin"
+	desc = "A safety pin for a dangerous weapon."
+	icon = 'icons/obj/guns/projectile.dmi'
+	icon_state = "SLRW-pin"
+	w_class = WEIGHT_CLASS_TINY
+	slot_flags = SLOT_EARS
+	
+/obj/item/SLRWpin/attack_self(mob/user)
+	if(istype(user.get_inactive_hand(), /obj/item/gun/projectile/S72LRW))
+		var/obj/item/gun/projectile/S72LRW/S = user.get_inactive_hand()
+		S.attackby(src, user)
+	
+/obj/item/gun/projectile/S72LRW/process_fire(atom/target as mob|obj|turf, mob/living/user as mob|obj, message = 1, params, zone_override = "")
```

Comment:
You shouldn't write `as X|Y|Z` in your arguments. I see that the base `process_fire` proc has these, and it probably autocreated them for you, which is misleading. This is, as I understand, legacy code that was done for... some reason in the past, but its really not needed anymore.

Also in proc arguments, if you need to have a boolean, such as for the `message` arg here, use `TRUE` and `FALSE` instead of 1/0. Again I see the base proc is using 1s and 0s as well, but just FYI, this is preferred.

Reply:
Same as last PR
`mob/living/user as mob|obj`

This isnt even possible